src/core/AutoRest.Core/CodeGenerator.cs

```diff
@@ -117,24 +117,30 @@ private void ResetFileList()
                     throw new Exception($"Duplicate File Generation: {filePath}");
                 }
                 FileList.Add(filePath);
-                    Settings.Instance.FileSystem.DeleteFile(filePath);
+                Settings.Instance.FileSystem.DeleteFile(filePath);
             }
             // Make sure the directory exist
             Settings.Instance.FileSystem.CreateDirectory(Path.GetDirectoryName(filePath));
 
+            var lineEnding = filePath.LineEnding();
+
```

Comment:
on the other hand, just saw below that we're also using `StringBuilder` and stuff, that's not customizable AFAIK... may just as well be explicit everywhere